### DISCUSSION TITLE
Decide if `EventReader::clear()` should return a `bool`

### DISCUSSION THREAD

**Discussion by IceSentry** [bevyengine/bevy#4707] [Polls]
We need to determine which option is best for #4693 

Option 1.

Clear doesn't return anything and you need to use `.is_empty()`

```rust
fn system(events: EventReader<MyEvent>) {
	if !events.is_empty {
		events.clear();
		// Process the fact that one or more event was received
	}
}
```

Option 2.

It returns whether or not there was any events to clear.

```rust
fn system(events: EventReader<MyEvent>) {
	if events.clear() {
		// Process the fact that one or more event was received
	}
}
```

**Comment by mockersf** [comment]
`clear()` is often used in std https://doc.rust-lang.org/stable/std/?search=clear and doesn't return anything

**Comment by Evrey** [comment]
I’d not be against just adding a third method in the spirit of staying with `libstd` convention. Such as: `checked_clear`, `clear_and_is_empty`, `maybe_clear`,…

**Reply by tigregalis** [reply]
In the same vein, I suggest `clear_then`.
```rust
fn system(events: EventReader<MyEvent>) {
	if events.clear_then() {
		// Process the fact that one or more event was received
	}
}
```

**Reply by Evrey** [reply]
By `libstd` conventions, `_then` would take a lambda.

**Reply by IceSentry** [reply]
Just letting you know, the PR is already merged. We decided to make clear() not return anything. You can open a new PR if you want, but this discussion is essentially closed, I'm just not sure how to close a discussion

**Reply by Evrey** [reply]
Ah, fun. =D
